### ISSUE TITLE
Improve FTB tracking in Windows multi user environment

### ISSUE THREAD

**Issue by topoguide** [MultiMC/MultiMC5#1093]
I ran MultiMC off of the main admin account on my Windows 7 PC, and created my first instance with this user account.  Every thing worked fine on that user account.  All other user accounts are unable to access that instance, and are unable to create a new instance (applies to both admin and user accounts).

This is how I set up MultiMC:
I unzipped MultiMC to a location in the Program Files folder so that all users had access to it.  I created a shortcut to MultiMC.exe and placed a copy on each users desktop.  I then ran MultiMC and  set up the first instance on the main admin account.  

The problem:
MultiMC is billed as portable, but it looks back at the original users profile data to run.  Because of this only the original user can run or create instances.  I have included a screen shot, and I have uploaded the log.   

![multmc_usererror](https://cloud.githubusercontent.com/assets/13058689/8370064/d94e10a8-1b77-11e5-9cd6-e4d8c83df9c2.png)

**Comment by Kilobyte22** [comment]
From what I remember MultiMC has a command line switch for setting run directory. See `MultiMC.exe --help`. Otherwise just set it's working directory. You may need a wrapper script for that

**Comment by Heufneutje** [comment]
Program Files might also not be the best location to install a portable application, since that sometimes requires special permissions.

**Comment by 02JanDal** [comment]
MultiMC is portable in the sense that it puts most data into it's own directory (it doesn't put stuff in %APPDATA% for example. This means that that installing into _Program Files_ it will use the same data for all users. The path to the user you're seeing in the settings is just to the FTB dir, which is determined from the %APPDATA% path, but that is also the only thing that uses paths outside of the MultiMC directory in any way.

You have two options, either (as @Kilobyte22 said) use the `-d <dir>` switch and set `<dir>` to be different for each user or (less hacky and what I'd recommend) install a separate copy of MultiMC for each user.

**Comment by peterix** [comment]
MultiMC does not belong in `Program Files`, same as it doesn't belong in `/usr/bin/` on linux. If it works at all, even for one user, it's by chance.

**Comment by topoguide** [comment]
Kilobyte - Thank you, I will try that.

02JanDal - the problem is that MultiMC is writing to the %APPDATA% path of the original user.  Which in my opinion makes it not-portable.  Hopefully with the suggested fix it will work.

Heufneutje - I thought of that, but I wanted it available to all users.  Instead of creating a seperate directory on the root Windows will allow you to set permissions so that all users can write to that specific folder.

Petrix - The comparison is not similar.  /usr/bin/ on linux is not for "portable" applications because they need to write back to their own folder, whereas Windows allows for that condition and many sys admins use this "feature" to not create multiple folders off of the root when one wants to set up a single instance of a program for all users.  Note: Linux has this built into the file structure with /opt, Windows has no structure for this.

**Comment by robotbrain** [comment]
the original user part is likely due to os permissions.
and if multimc writes to appdata for anything other than temp, its a bug

**Comment by topoguide** [comment]
Found the problem.  The default setting for MultiMC in the multimc.cfg assigns the ftb launcher the original user as the appdata folder to use for all instances.

What I did is substitute the following code into multimc.cfg
      FTBLauncherRoot=C:/Users/$USERNAME/AppData/Roaming/ftblauncher

It seems to work flawlessly for all users.  -- Perhaps MultiMC could be updated to have this be the default verbage in the .cfg file.  This IMO would make MultiMC more of a portable structure.

Thanks for everyone's help!
